src/Ide/Plugin/Tactic/Machinery.hs

```diff
@@ -0,0 +1,259 @@
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MonoLocalBinds        #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeSynonymInstances  #-}
+{-# LANGUAGE ViewPatterns          #-}
+
+module Ide.Plugin.Tactic.Machinery where
+
+import           Control.Arrow
+import           Control.Monad.State (get, modify, evalStateT)
+import           Data.Char
+import           Data.Function
+import           Data.List
+import           Data.Map (Map)
+import qualified Data.Map as M
+import           Data.Maybe
+import           Data.Set (Set)
+import qualified Data.Set as S
+import           Data.Traversable
+import           DataCon
+import           Development.IDE.Types.Location
+import           DynFlags (unsafeGlobalDynFlags)
+import qualified FastString as FS
+import           GHC
```

Comment:
Prefer `Development.IDE.GHC.Compat`

Reply:
Are the other GHC modules ok? Like FastString, DataCon and OccName?

Reply:
Yeah